Issue 1051: Add current environment docs and diagrams (#1070)
raft-tech/TANF-app: docs/Technical-Documentation/TDP-environments-README.md

# TDP Environments

## Development

| Dev Site | Frontend URL | Backend URL | Purpose |
| -------- | -------- | -------- | -------- |
| Sandbox     | https://tdp-frontend-sandbox.app.cloud.gov | https://tdp-backend-sandbox.app.cloud.gov     | Space for devs to test in a deployed environment 
| A11y | https://tdp-frontend-a11y.app.cloud.gov | https://tdp-backend-a11y.app.cloud.gov | Space for accessibility testing |
| QASP | https://tdp-frontend-qasp.app.cloud.gov | https://tdp-backend-qasp.app.cloud.gov | Space for QASP review |
| raft | https://tdp-frontend-raft.app.cloud.gov | https://tdp-backend-raft.app.cloud.gov | Space for raft review |

### Common Dependencies

**Cloud.gov AWS RDS `(tanf-dev)`**
- `tdp-db-dev`
  
**Cloud.gov AWS S3 `(tanf-dev)`**
- `tdp-staticfiles-dev`
- `tdp-datafiles-dev`
- `tdp-tf-states` - Stores the Terraform state files used to create and re-recreate services infrastructure.

> 

Reviewer: I would add a dependency for clamav-rest to each of the sections here as well as in the diagram. There will be one instance of ClamAV per cloud.gov space.

# TDP Environments

## Development

| Dev Site | Frontend URL | Backend URL | Purpose |
| -------- | -------- | -------- | -------- |

> | Sandbox     | https://tdp-frontend-sandbox.app.cloud.gov | https://tdp-backend-sandbox.app.cloud.gov     | Space for devs to test in a deployed environment 

Reviewer 2: ```suggestion
| Sandbox     | https://tdp-frontend-sandbox.app.cloud.gov | https://tdp-backend-sandbox.app.cloud.gov/admin/     | Space for devs to test in a deployed environment 
```

> | A11y | https://tdp-frontend-a11y.app.cloud.gov | https://tdp-backend-a11y.app.cloud.gov | Space for accessibility testing |

Reviewer 2: ```suggestion
| A11y | https://tdp-frontend-a11y.app.cloud.gov | https://tdp-backend-a11y.app.cloud.gov/admin/ | Space for accessibility testing |
```

> | QASP | https://tdp-frontend-qasp.app.cloud.gov | https://tdp-backend-qasp.app.cloud.gov | Space for QASP review |

Reviewer 2: ```suggestion
| QASP | https://tdp-frontend-qasp.app.cloud.gov | https://tdp-backend-qasp.app.cloud.gov/admin/ | Space for QASP review |
```

> | raft | https://tdp-frontend-raft.app.cloud.gov | https://tdp-backend-raft.app.cloud.gov | Space for raft review |

Reviewer 2: ```suggestion
| raft | https://tdp-frontend-raft.app.cloud.gov | https://tdp-backend-raft.app.cloud.gov/admin/ | Space for raft review |
```

### Dependencies 

- `clamav-rest`

**Cloud.gov AWS RDS `(tanf-dev)`**

> - `tdp-db-dev`

Reviewer 2: will any parsed test data be stored here? If so, could include this in description of what's here?

Author: > not from unit tests - that data gets deleted on test teardown. this database stores users, groups, stts, metadata about data files that have been submitted, etc (Django data models). It has not been decided if parsed data will end up in this DB or another service

Author: Test data is not stored here; this DB will primarily contain the Users, Reports, and other application-related models.

  
**Cloud.gov AWS S3 `(tanf-dev)`**

> - `tdp-staticfiles-dev`

Reviewer 2: worth including what's stored in here?

> - `tdp-datafiles-dev`

Reviewer 2: any sample raw data files to be stored here?

Author: I will clarify in the README – `datafiles` will store uploaded files from STTs. Staticfiles will stre Django admin HTML/CSS.

> - `tdp-datafiles-dev` - Stores the files uploaded by STTs.

Reviewer 2: ```suggestion
- `tdp-datafiles-dev` - Stores the files uploaded by STTs (no real STT data to be stored in dev).
```

- `tdp-tf-states` - Stores the Terraform state files used to create and re-recreate services infrastructure.

## Staging

Unlike Development, the Staging environment contains a single frontend and backend deployment.

| Frontend URL | Backend URL | Purpose |
| -------- | -------- | -------- |

> | https://tdp-frontend-staging.app.cloud.gov | https://tdp-backend-staging.app.cloud.gov     | Space for government users to test in a deployed, production-like environment    |

Reviewer 2: ```suggestion
| https://tdp-frontend-staging.app.cloud.gov | https://tdp-backend-staging.app.cloud.gov/admin/     | Space for government users to test in a deployed, production-like environment    |
```

### Dependencies 

- `clamav-rest` - Virus scanner REST service used to scan file uploads. 

**Cloud.gov AWS RDS `(tanf-staging)`**
- `tdp-db-staging` - Stores application-level models (e.g. Users, Reports).
  
**Cloud.gov AWS S3 `(tanf-staging)`**
- `tdp-staticfiles-staging` - Stores static HTML/CSS for Django Admin.

> - `tdp-datafiles-staging` - Stores the files uploaded by STTs.

Reviewer 2: ```suggestion
- `tdp-datafiles-staging` - Stores the files uploaded by STTs (no real STT data to be stored in staging).
```

  
**Cloud.gov AWS S3 `(tanf-dev)`**
- `tdp-tf-states` - Stores the Terraform state files used to create and re-recreate services infrastructure. Note this S3 bucket lives in the development space.

## Production

Like Staging, there is only one Production deployment.

| Frontend URL | Backend URL | Purpose |
| -------- | -------- | -------- |

> | https://tdp-frontend-production.app.cloud.gov | https://tdp-backend-production.app.cloud.gov     | Production space for active users of the application.    |

Reviewer 2: pretty sure these will change [ref](https://github.com/raft-tech/TANF-app/issues/726#issuecomment-801111466) --- maybe `tbd` until we confirm it.

Author: ```suggestion
| `tbd`` | `tbd`     | Production space for active users of the application.    |
```
